velox/vector/FlatVector.cpp

```diff
@@ -281,14 +281,44 @@ void FlatVector<StringView>::copy(
     copyValuesAndNulls(source, rows, toSourceRow);
     acquireSharedStringBuffers(source);
   } else {
+    DecodedVector decoded(*source);
```

Comment:
I am curious why using decoded vector here help performance? Thanks!

Reply:
It helps avoid virtual function calls to check isNullAt(sourceRow) and to get valueAt(sourceRow).